src/main/java/com/exasol/dbbuilder/objectwriter/ImmediateDatabaseObjectWriter.java

```diff
@@ -0,0 +1,182 @@
+                preparedStatement.setObject(i, parameters[i]);
+            }
+            preparedStatement.execute();
+        } catch (final SQLException exception) {
+            throw new DatabaseObjectException(object, "Failed to write to object: " + sql, exception);
+        }
+    }
+
+    @Override
+    public void write(final ConnectionDefinition definition) {
+        if (definition.hasUserName()) {
+            if (definition.hasPassword()) {
+                writeToObject(definition,
+                        "CREATE CONNECTION " + definition.getFullyQualifiedName() + " TO '" + definition.getTo()
+                                + "' USER '" + definition.getUserName() + ("' IDENTIFIED BY '")
+                                + definition.getPassword() + "'");
+            } else {
+                throw new DatabaseObjectException(definition,
+                        "Password missing when trying to write connection definition "
+                                + definition.getFullyQualifiedName()
+                                + ". Please alway provide user name and password together or not at all.");
+            }
+        } else {
+            if (definition.hasPassword()) {
+                throw new DatabaseObjectException(definition,
+                        "User name missing when trying to write connection definition "
+                                + definition.getFullyQualifiedName()
+                                + ". Please alway provide user name and password together or not at all.");
+            } else {
+                writeToObject(definition,
+                        "CREATE CONNECTION " + definition.getFullyQualifiedName() + " TO '" + definition.getTo() + "'");
+            }
+        }
+    }
+
+    @Override
+    public void write(final Schema schema) {
+        writeToObject(schema, "CREATE SCHEMA " + schema.getFullyQualifiedName());
+    }
+
+    @Override
+    public void write(final Table table) {
+        final StringBuilder sqlBuilder = new StringBuilder("CREATE TABLE " + table.getFullyQualifiedName() + " (");
+        boolean first = true;
+        for (final Column column : table.getColumns()) {
+            if (first) {
+                first = false;
+            } else {
+                sqlBuilder.append(", ");
+            }
+            sqlBuilder.append("\"");
+            sqlBuilder.append(column.getName());
+            sqlBuilder.append("\" ");
+            sqlBuilder.append(column.getType());
+        }
+        sqlBuilder.append(")");
+        writeToObject(table, sqlBuilder.toString());
+    }
+
+    @Override
+    public void write(final Table table, final Object... values) {
+        final String valuePlaceholders = "?" + ", ?".repeat(table.getColumnCount() - 1);
+        try (final PreparedStatement insert = this.connection.prepareStatement(
+                "INSERT INTO " + table.getFullyQualifiedName() + " VALUES(" + valuePlaceholders + ")");) {
+            int columnNumber = 1;
```

Comment:
In this function, you can call the `writeToObject` function with parameters.

Reply:
Rewritten.